### ISSUE TITLE
[Feature Request] Support gdbserver debugging (miDebuggerServerAddress) when pipeTransport is used 

### ISSUE THREAD

**Issue by Jackie-Yang** [microsoft/vscode-cpptools#528]
Our project is very huge, so we connect to our server using ssh, and use gdb on our server (our project is on our server too), now I can use gdb on my computer to debug, but it is to slow, can I use gdb on my server to debug?

**Comment by pieandcakes** [comment]
@Jackie-Yang please see https://github.com/Microsoft/vscode-cpptools/blob/master/Documentation/Debugger/gdb/PipeTransport.md

**Comment by Jackie-Yang** [comment]
@pieandcakes Wow, it's very useful! But it seems that "miDebuggerServerAddress" can not use after I set the PipeTranspor??? Is that I can't use them at the same time?
Before I set the PipeTranspor
![before](https://cloud.githubusercontent.com/assets/18381153/23708688/c3fe0c02-0451-11e7-83b2-4c4a740ea8b4.PNG)
After I set the PipeTranspor
![after](https://cloud.githubusercontent.com/assets/18381153/23708715/dd228956-0451-11e7-96e3-cb8b132e3db3.PNG)

**Comment by pieandcakes** [comment]
@Jackie-Yang Why would you ssh to a remote gdb and then connect to a gdbserver instead of just using ssh to connect to the final instance of gdb? This isn't a scenario that is currently supported but if you can explain why you want it, we can add it to the backlog.

**Comment by Jackie-Yang** [comment]
@pieandcakes My program is running on ARM ,and my code(almost 20G) is on server, so I need to debug my program by using gdbserver, I used to mount server to my PC, then use gdbserver to debug ARM program, but it is slower than using gdbserve on server, so I want to use pipetranspor to connect server, and debug ARM program by using gdbserver.

**Comment by Jackie-Yang** [comment]
@pieandcakes Did I make it clear ?

**Comment by pieandcakes** [comment]
@Jackie-Yang Yes. I'll mark it as an enhancement and add it to the backlog. Thanks

**Comment by Jyoti12393** [comment]
@pieandcakes can you tell me step by step to compile c++ vs project on server using ssh.

**Comment by Jackie-Yang** [comment]
@sean-mcmanus @pieandcakes 
Hello~
Has this feature been realized?

**Comment by pieandcakes** [comment]
@Jackie-Yang We haven't had time to investigate and resolve this. The code that does the work is open sourced. The program that handles the commands is [MIEngine](https://github.com/microsoft/miengine) if you would like to investigate a fix and contribute. Otherwise, you can try using a combination of `setupCommands` and `pipeTransport` and see if that is a workaround for you.

**Comment by muekai1** [comment]
Hi,
Szenario for which a remote debugger on a device plus a pipecommand gdb  is needed:
Build in a WSL or Container and Debug on a Device.

Developement for "embeded" Linux devices:
You cannot compile/build on the device. So this is done in a wsl/ container/ buildmachine.
You cannot debug hardware depend code in a subsystem so debuging needs to happen on the real device,
Resources on some devices are limited which makes transfering debug symbols ond source to the device difficult.

So what would work is using a gdb in wsl/container as midebugger via pipe as it has access debug symbols as well as source and start the binary on the remote server.

[ Transfering linux stuff back to Windows and use a windows mingw gdb would work. However linendings and (in general)caseinsensitive filesystem lead to surprising results. Using wsl and pointing to the Windows mappping into wsl may work too but freezes due to slow wsl<->windows filesystem integration]

Ultimately it realy makes sense to start the debugger where the build happens and debug on the real device.

**Comment by muekai1** [comment]
does not work.  ip or name is not the problem but variables are get messed up/are not defined. also "debuggerConfiguration" is ignored when using pipecommand. 

and besides: connectionmanager? it is supposed to work with connectionmanager and ${defaultRemoteMachineName} should hold all information - if it wern't ignore. and so on and so on....
